gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/src/AuthServer.js

```diff
@@ -506,4 +515,29 @@ gpii.oauth2.authServer.contributeRouteHandlers = function (that, oauth2orizeServ
         }
     );
 
+    that.expressApp.post("/add-preferences",
+        function (req, res) {
+            var accessToken = gpii.oauth2.parseBearerAuthorizationHeader(req);
+            if (!accessToken) {
+                res.send(401);
+            } else {
+                var client = that.authorizationService.getClientByClientCredentialsAccessToken(accessToken);
+                if (!client) {
+                    res.send(404);
+                }
+
+                var tokenPrivs = that.authorizationService.getClientCredentialsTokenPrivs(accessToken);
+                if (!tokenPrivs.allowAddPrefs) {
+                    res.send(401);
+                } else {
+                    var savePrefsPromise = that.authorizationService.savePrefs(req.body, req.query.view);
+                    savePrefsPromise.then(function (data) {
+                        res.json(data);
+                    }, function () {
```

Comment:
There may be an error argument here which we should propagate in the response body